File: shein/src/main/java/com/shein/Models/BaseReturn.java
package com.shein.Models;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
public class BaseReturn {

    private String _message;
    private boolean _worked;
    private Object _Object;

    public BaseReturn Success(String message){
        return new BaseReturn(
            message,
            true
        );
    }
    public BaseReturn Error(String message){
        return new BaseReturn(
            message,
            false
        );
    }
    public BaseReturn(String _message, boolean _worked) {
        this._message = _message;
        this._worked = _worked;
    }
    public BaseReturn Success(String message, Object object){
        return new BaseReturn(
            message,
            _worked = true,
            _Object = object
        );
    }
    
}

File: api.aliexpress/Domain/Core/Base/BaseReturn.cs
﻿using Domain.Core.Base;
using System.Net;

public record BaseReturn
{
    public object? Body;
    public BaseError? Error;
    public HttpStatusCode StatusCode = HttpStatusCode.OK;

    public BaseReturn()
    {

    }

    public void HttpSuccess<TBody>(TBody successObject)
    {
        Body = successObject;
    }


    public BaseReturn Success(object successObject)
    {
        Body = successObject;

        return this;
    }


    public BaseReturn ErrorBussiness(string message)
    {
        StatusCode = HttpStatusCode.BadRequest;

        Error = new()
        {
            code = "400",
            message = $"{message}",
        };

        return this;
    }
    public BaseReturn ErrorSystem(Exception ex)
    {
        StatusCode = HttpStatusCode.InternalServerError;

        Error = new()
        {
            code = "500",
            message = $"System Error: {ex.Message}",
            info = ex.StackTrace,
        };

        return this;
    }

    public object? GetBody()
    {
        if (StatusCode is HttpStatusCode.OK) return Body;
        return Error;
    }

    public IResult GetResponse()
    {
        if (StatusCode is HttpStatusCode.OK) return Results.Ok(Body);
        return Results.Json(Error, statusCode: (int)StatusCode);
    }
}